teamcity: failed tests on master: test/TestSystemConfigGossip

The following tests appear to have failed:

[#90756](https://teamcity.cockroachdb.com/viewLog.html?buildId=90756):

```
--- FAIL: test/TestSystemConfigGossip (0.009s)
	server_test.go:435: unexpected message received on gossip channel: {[{/Table/2/1/0/"system"/3/1 {[18 88 97 62 1 2] 1481747135.636388217,0}} {/Table/2/1/1/"descriptor"/3/1 {[136 238 172 209 1 6] 1481747135.636388217,0}} {/Table/2/1/1/"eventlog"/3/1 {[44 21 140 5 1 24] 1481747135.636388217,0}} {/Table/2/1/1/"lease"/3/1 {[205 248 213 252 1 22] 1481747135.636388217,0}} {/Table/2/1/1/"namespace"/3/1 {[60 185 31 84 1 4] 1481747135.636388217,0}} {/Table/2/1/1/"rangelog"/3/1 {[202 163 171 90 1 26] 1481747135.636388217,0}} {/Table/2/1/1/"ui"/3/1 {[123 247 193 129 1 28] 1481747135.636388217,0}} {/Table/2/1/1/"users"/3/1 {[93 180 27 24 1 8] 1481747135.636388217,0}} {/Table/2/1/1/"zones"/3/1 {[233 251 73 123 1 10] 1481747135.636388217,0}} {/Table/3/1/1/2/1 {[188 24 237 68 3 18 22 10 6 115 121 115 116 101 109 16 1 26 10 10 8 10 4 114 111 111 116 16 48] 1481747135.636388217,0}} {/Table/3/1/2/2/1 {[160 181 196 70 3 10 234 1 10 9 110 97 109 101 115 112 97 99 101 24 2 32 1 40 1 48 0 58 4 8 0 16 0 66 24 10 8 112 97 114 101 110 116 73 68 16 1 26 6 8 1 16 0 24 0 32 0 48 0 66 20 10 4 110 97 109 101 16 2 26 6 8 7 16 0 24 0 32 0 48 0 66 18 10 2 105 100 16 3 26 6 8 1 16 0 24 0 32 1 48 0 72 4 82 49 10 7 112 114 105 109 97 114 121 16 1 24 1 34 8 112 97 114 101 110 116 73 68 34 4 110 97 109 101 48 1 48 2 64 0 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 10 10 8 10 4 114 111 111 116 16 48 128 1 1 136 1 3 152 1 0 178 1 33 10 7 112 114 105 109 97 114 121 16 0 26 8 112 97 114 101 110 116 73 68 26 4 110 97 109 101 32 1 32 2 40 0 178 1 20 10 8 102 97 109 95 51 95 105 100 16 3 26 2 105 100 32 3 40 3 184 1 4 194 1 0] 1481747135.636388217,0}} {/Table/3/1/3/2/1 {[219 94 158 34 3 10 201 1 10 10 100 101 115 99 114 105 112 116 111 114 24 3 32 1 40 1 48 0 58 4 8 0 16 0 66 18 10 2 105 100 16 1 26 6 8 1 16 0 24 0 32 0 48 0 66 26 10 10 100 101 115 99 114 105 112 116 111 114 16 2 26 6 8 8 16 0 24 0 32 1 48 0 72 3 82 33 10 7 112 114 105 109 97 114 121 16 1 24 1 34 2 105 100 48 1 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 10 10 8 10 4 114 111 111 116 16 48 128 1 1 136 1 3 152 1 0 178 1 19 10 7 112 114 105 109 97 114 121 16 0 26 2 105 100 32 1 40 0 178 1 36 10 16 102 97 109 95 50 95 100 101 115 99 114 105 112 116 111 114 16 2 26 10 100 101 115 99 114 105 112 116 111 114 32 2 40 2 184 1 3 194 1 0] 1481747135.636388217,0}} {/Table/3/1/4/2/1 {[112 93 162 105 3 10 227 1 10 5 117 115 101 114 115 24 4 32 1 40 1 48 0 58 4 8 0 16 0 66 24 10 8 117 115 101 114 110 97 109 101 16 1 26 6 8 7 16 0 24 0 32 0 48 0 66 30 10 14 104 97 115 104 101 100 80 97 115 115 119 111 114 100 16 2 26 6 8 8 16 0 24 0 32 1 48 0 72 3 82 39 10 7 112 114 105 109 97 114 121 16 1 24 1 34 8 117 115 101 114 110 97 109 101 48 1 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 11 10 9 10 4 114 111 111 116 16 240 3 128 1 1 136 1 3 152 1 0 178 1 25 10 7 112 114 105 109 97 114 121 16 0 26 8 117 115 101 114 110 97 109 101 32 1 40 0 178 1 44 10 20 102 97 109 95 50 95 104 97 115 104 101 100 80 97 115 115 119 111 114 100 16 2 26 14 104 97 115 104 101 100 80 97 115 115 119 111 114 100 32 2 40 2 184 1 3 194 1 0] 1481747135.636388217,0}} {/Table/3/1/5/2/1 {[49 110 61 249 3 10 185 1 10 5 122 111 110 101 115 24 5 32 1 40 1 48 0 58 4 8 0 16 0 66 18 10 2 105 100 16 1 26 6 8 1 16 0 24 0 32 0 48 0 66 22 10 6 99 111 110 102 105 103 16 2 26 6 8 8 16 0 24 0 32 1 48 0 72 3 82 33 10 7 112 114 105 109 97 114 121 16 1 24 1 34 2 105 100 48 1 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 11 10 9 10 4 114 111 111 116 16 240 3 128 1 1 136 1 3 152 1 0 178 1 19 10 7 112 114 105 109 97 114 121 16 0 26 2 105 100 32 1 40 0 178 1 28 10 12 102 97 109 95 50 95 99 111 110 102 105 103 16 2 26 6 99 111 110 102 105 103 32 2 40 2 184 1 3 194 1 0] 1481747135.636388217,0}} {/Table/3/1/11/2/1 {[96 37 19 216 3 10 166 2 10 5 108 101 97 115 101 24 11 32 1 40 1 48 0 58 4 8 0 16 0 66 22 10 6 100 101 115 99 73 68 16 1 26 6 8 1 16 0 24 0 32 0 48 0 66 23 10 7 118 101 114 115 105 111 110 16 2 26 6 8 1 16 0 24 0 32 0 48 0 66 22 10 6 110 111 100 101 73 68 16 3 26 6 8 1 16 0 24 0 32 0 48 0 66 26 10 10 101 120 112 105 114 97 116 105 111 110 16 4 26 6 8 5 16 0 24 0 32 0 48 0 72 5 82 78 10 7 112 114 105 109 97 114 121 16 1 24 1 34 6 100 101 115 99 73 68 34 7 118 101 114 115 105 111 110 34 10 101 120 112 105 114 97 116 105 111 110 34 6 110 111 100 101 73 68 48 1 48 2 48 4 48 3 64 0 64 0 64 0 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 10 10 8 10 4 114 111 111 116 16 2 128 1 1 136 1 3 152 1 0 178 1 58 10 7 112 114 105 109 97 114 121 16 0 26 6 100 101 115 99 73 68 26 7 118 101 114 115 105 111 110 26 6 110 111 100 101 73 68 26 10 101 120 112 105 114 97 116 105 111 110 32 1 32 2 32 3 32 4 40 0 184 1 1 194 1 0] 1481747135.636388217,0}} {/Table/3/1/12/2/1 {[152 17 124 55 3 10 212 3 10 8 101 118 101 110 116 108 111 103 24 12 32 1 40 2 48 0 58 12 8 254 217 132 191 181 165 142 200 20 16 0 66 25 10 9 116 105 109 101 115 116 97 109 112 16 1 26 6 8 5 16 0 24 0 32 0 48 0 66 25 10 9 101 118 101 110 116 84 121 112 101 16 2 26 6 8 7 16 0 24 0 32 0 48 0 66 24 10 8 116 97 114 103 101 116 73 68 16 3 26 6 8 1 16 0 24 0 32 0 48 0 66 27 10 11 114 101 112 111 114 116 105 110 103 73 68 16 4 26 6 8 1 16 0 24 0 32 0 48 0 66 20 10 4 105 110 102 111 16 5 26 6 8 7 16 0 24 0 32 1 48 0 66 35 10 8 117 110 105 113 117 101 73 68 16 6 26 6 8 8 16 0 24 0 32 0 42 9 117 117 105 100 95 118 52 40 41 48 0 72 7 82 54 10 7 112 114 105 109 97 114 121 16 1 24 1 34 9 116 105 109 101 115 116 97 109 112 34 8 117 110 105 113 117 101 73 68 48 1 48 6 64 0 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 10 10 8 10 4 114 111 111 116 16 2 128 1 1 136 1 3 152 1 0 178 1 38 10 7 112 114 105 109 97 114 121 16 0 26 9 116 105 109 101 115 116 97 109 112 26 8 117 110 105 113 117 101 73 68 32 1 32 6 40 0 178 1 34 10 15 102 97 109 95 50 95 101 118 101 110 116 84 121 112 101 16 2 26 9 101 118 101 110 116 84 121 112 101 32 2 40 2 178 1 32 10 14 102 97 109 95 51 95 116 97 114 103 101 116 73 68 16 3 26 8 116 97 114 103 101 116 73 68 32 3 40 3 178 1 38 10 17 102 97 109 95 52 95 114 101 112 111 114 116 105 110 103 73 68 16 4 26 11 114 101 112 111 114 116 105 110 103 73 68 32 4 40 4 178 1 24 10 10 102 97 109 95 53 95 105 110 102 111 16 5 26 4 105 110 102 111 32 5 40 5 184 1 6 194 1 0] 1481747135.730118653,0}} {/Table/3/1/13/2/1 {[74 81 41 195 3 10 139 4 10 8 114 97 110 103 101 108 111 103 24 13 32 1 40 1 48 0 58 4 8 0 16 0 66 25 10 9 116 105 109 101 115 116 97 109 112 16 1 26 6 8 5 16 0 24 0 32 0 48 0 66 23 10 7 114 97 110 103 101 73 68 16 2 26 6 8 1 16 0 24 0 32 0 48 0 66 23 10 7 115 116 111 114 101 73 68 16 3 26 6 8 1 16 0 24 0 32 0 48 0 66 25 10 9 101 118 101 110 116 84 121 112 101 16 4 26 6 8 7 16 0 24 0 32 0 48 0 66 28 10 12 111 116 104 101 114 82 97 110 103 101 73 68 16 5 26 6 8 1 16 0 24 0 32 1 48 0 66 20 10 4 105 110 102 111 16 6 26 6 8 7 16 0 24 0 32 1 48 0 66 40 10 8 117 110 105 113 117 101 73 68 16 7 26 6 8 1 16 0 24 0 32 0 42 14 117 110 105 113 117 101 95 114 111 119 105 100 40 41 48 0 72 8 82 54 10 7 112 114 105 109 97 114 121 16 1 24 1 34 9 116 105 109 101 115 116 97 109 112 34 8 117 110 105 113 117 101 73 68 48 1 48 7 64 0 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 10 10 8 10 4 114 111 111 116 16 2 128 1 1 136 1 3 152 1 0 178 1 38 10 7 112 114 105 109 97 114 121 16 0 26 9 116 105 109 101 115 116 97 109 112 26 8 117 110 105 113 117 101 73 68 32 1 32 7 40 0 178 1 30 10 13 102 97 109 95 50 95 114 97 110 103 101 73 68 16 2 26 7 114 97 110 103 101 73 68 32 2 40 2 178 1 30 10 13 102 97 109 95 51 95 115 116 111 114 101 73 68 16 3 26 7 115 116 111 114 101 73 68 32 3 40 3 178 1 34 10 15 102 97 109 95 52 95 101 118 101 110 116 84 121 112 101 16 4 26 9 101 118 101 110 116 84 121 112 101 32 4 40 4 178 1 40 10 18 102 97 109 95 53 95 111 116 104 101 114 82 97 110 103 101 73 68 16 5 26 12 111 116 104 101 114 82 97 110 103 101 73 68 32 5 40 5 178 1 24 10 10 102 97 109 95 54 95 105 110 102 111 16 6 26 4 105 110 102 111 32 6 40 6 184 1 7 194 1 0] 1481747135.636388217,0}} {/Table/3/1/14/2/1 {[188 163 39 71 3 10 251 1 10 2 117 105 24 14 32 1 40 1 48 0 58 4 8 0 16 0 66 19 10 3 107 101 121 16 1 26 6 8 7 16 0 24 0 32 0 48 0 66 21 10 5 118 97 108 117 101 16 2 26 6 8 8 16 0 24 0 32 1 48 0 66 27 10 11 108 97 115 116 85 112 100 97 116 101 100 16 3 26 6 8 5 16 0 24 0 32 0 48 0 72 4 82 34 10 7 112 114 105 109 97 114 121 16 1 24 1 34 3 107 101 121 48 1 64 0 74 8 8 0 16 0 26 0 32 0 90 0 96 2 106 10 10 8 10 4 114 111 111 116 16 2 128 1 1 136 1 3 152 1 0 178 1 20 10 7 112 114 105 109 97 114 121 16 0 26 3 107 101 121 32 1 40 0 178 1 26 10 11 102 97 109 95 50 95 118 97 108 117 101 16 2 26 5 118 97 108 117 101 32 2 40 2 178 1 38 10 17 102 97 109 95 51 95 108 97 115 116 85 112 100 97 116 101 100 16 3 26 11 108 97 115 116 85 112 100 97 116 101 100 32 3 40 3 184 1 4 194 1 0] 1481747135.636388217,0}} {/Table/5/1/0/2/1 {[40 49 224 151 3 16 128 128 64 24 128 128 128 32 34 4 8 128 163 5 40 1 50 0] 1481747135.636388217,0}}]}

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/server/server_test.go
+++ b/pkg/server/server_test.go
@@ -49,6 +49,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/metric"
 	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/gogo/protobuf/jsonpb"
+	"github.com/pkg/errors"
 )
 
 var nodeTestBaseContext = testutils.NewNodeTestBaseContext()
@@ -413,33 +414,7 @@ func TestSystemConfigGossip(t *testing.T) {
 		t.Fatal("did not receive gossip message")
 	}
 
-	// Try a plain KV write first.
-	if err := kvDB.Put(ctx, key, valAt(0)); err != nil {
-		t.Fatal(err)
-	}
-
-	// Now do it as part of a transaction, but without the trigger set.
-	if err := kvDB.Txn(ctx, func(txn *client.Txn) error {
-		return txn.Put(key, valAt(1))
-	}); err != nil {
-		t.Fatal(err)
-	}
-
-	// Gossip channel should be dormant.
-	// TODO(tschottdorf): This test is likely flaky. Why can't some other
-	// process trigger gossip? It seems that a new range lease being
-	// acquired will gossip a new system config since the hash changed and fail
-	// the test (seen in practice during some buggy WIP).
-	var systemConfig config.SystemConfig
-	select {
-	case <-resultChan:
-		systemConfig, _ = ts.gossip.GetSystemConfig()
-		t.Fatalf("unexpected message received on gossip channel: %v", systemConfig)
-
-	case <-time.After(50 * time.Millisecond):
-	}
-
-	// This time mark the transaction as having a Gossip trigger.
+	// Write a system key with the transaction marked as having a Gossip trigger.
 	if err := kvDB.Txn(ctx, func(txn *client.Txn) error {
 		txn.SetSystemConfigTrigger()
 		return txn.Put(key, valAt(2))
@@ -447,35 +422,42 @@ func TestSystemConfigGossip(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// New system config received.
-	select {
-	case <-resultChan:
-		systemConfig, _ = ts.gossip.GetSystemConfig()
+	// This has to be wrapped in a SucceedSoon because system migrations on the
+	// testserver's startup can trigger system config updates without the key we
+	// wrote.
+	testutils.SucceedsSoon(t, func() error {
+		// New system config received.
+		var systemConfig config.SystemConfig
+		select {
+		case <-resultChan:
+			systemConfig, _ = ts.gossip.GetSystemConfig()
 
-	case <-time.After(500 * time.Millisecond):
-		t.Fatal("did not receive gossip message")
-	}
-
-	// Now check the new config.
-	var val *roachpb.Value
-	for _, kv := range systemConfig.Values {
-		if bytes.Equal(key, kv.Key) {
-			val = &kv.Value
-			break
+		case <-time.After(500 * time.Millisecond):
+			return errors.Errorf("did not receive gossip message")
 		}
-	}
-	if val == nil {
-		t.Fatal("key not found in gossiped info")
-	}
 
-	// Make sure the returned value is valAt(2).
-	got := new(sqlbase.DatabaseDescriptor)
-	if err := val.GetProto(got); err != nil {
-		t.Fatal(err)
-	}
-	if expected := valAt(2); !reflect.DeepEqual(got, expected) {
-		t.Fatalf("mismatch: expected %+v, got %+v", *expected, *got)
-	}
+		// Now check the new config.
+		var val *roachpb.Value
+		for _, kv := range systemConfig.Values {
+			if bytes.Equal(key, kv.Key) {
+				val = &kv.Value
+				break
+			}
+		}
+		if val == nil {
+			return errors.Errorf("key not found in gossiped info")
+		}
+
+		// Make sure the returned value is valAt(2).
+		got := new(sqlbase.DatabaseDescriptor)
+		if err := val.GetProto(got); err != nil {
+			return err
+		}
+		if expected := valAt(2); !reflect.DeepEqual(got, expected) {
+			return errors.Errorf("mismatch: expected %+v, got %+v", *expected, *got)
+		}
+		return nil
+	})
 }
 
 func checkOfficialize(t *testing.T, network, oldAddrString, newAddrString, expAddrString string) {

